web/apps/apps.go

```diff
@@ -214,6 +223,46 @@ func installHandler(installerType consts.AppType) echo.HandlerFunc {
 	}
 }
 
+// logsHandler handles all POST /:slug/logs requests and forwards to sent log
+// lines to the server logger.
+func logsHandler(appType consts.AppType) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		inst := middlewares.GetInstance(c)
+
+		slug := c.Param("slug")
+		_, err := app.GetBySlug(inst, slug, appType)
+		if err != nil {
+			return wrapAppsError(err)
+		}
+
+		if err := middlewares.AllowWholeType(c, permission.POST, consts.AppLogs); err != nil {
+			return err
+		}
+
+		body, err := ioutil.ReadAll(c.Request().Body)
+		if err != nil {
+			return wrapAppsError(err)
+		}
+
+		var logs []AppLog
+		if err := json.Unmarshal(body, &logs); err != nil {
+			fmt.Println("Bad json: ", err)
+			return echo.NewHTTPError(http.StatusBadRequest)
+		}
+
+		logger := logger.WithDomain(inst.Domain).WithNamespace(slug)
+		for _, log := range logs {
+			level, err := logrus.ParseLevel(log.Level)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusBadRequest)
```

Comment:
I would prefer to have an explicit error which tells to the caller that it is the `level` that has this issue. You can use a JSON-API error for that:

```go
return jsonapi.InvalidAttribute("level", err)
```